lua/autorun/client/TrueCulling.lua

```diff
@@ -0,0 +1,68 @@
+jit.on()
+
+CULLFRONT = math.sin(0.57357644) -- Degrees forward angle in sine
+
+function main()
+	allEnts = ents.GetAll()
```

Comment:
So, you're calling ents.GetAll, a very expensive function, every single frame...

Reply:
You should be using the hooks EntityCreated/NetworkEntityCreated/EntityRemoved (if I remember those names correctly) to maintain a single global table of all entities. And the entities should be keys so you are not doing an O(N) table.remove when an entity is deleted.

Reply:
Also you need to be using local variables so it is not looking up the variable name in the global namespace each access